Don't use anonymous namespace in header.

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#21790 Don't use anonymous namespace in header.**

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D15827311](https://our.internmc.facebook.com/intern/diff/D15827311)

ezyang committed:
Don't use anonymous namespace in header.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang commented:
cc @kmatzen

colesbury commented on aten/src/ATen/core/PhiloxRNGEngine.h:
```diff
@@ -21,7 +21,7 @@
 
 // typedefs for holding vector data
-namespace {
+namespace detail {
```
There's still a comment "// anonymous namespace" below

colesbury commented on aten/src/ATen/core/PhiloxRNGEngine.h:
```diff
@@ -21,7 +21,7 @@
-namespace {
+namespace detail {
 
 typedef at::detail::Array<uint32_t, 4> UINT4;
```
It looks like these are only used in `philox_engine` so maybe they should just be placed in `class philox_engine`, but this is fine too

ezyang replied:
You're right, but I am a lazy man :)

ezyang committed:
Update on "Don't use anonymous namespace in header."

Don't use anonymous namespace in header.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 21790 gh/ezyang/172/head
